src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Extensions/PositionExtensions.cs

```diff
@@ -26,6 +26,16 @@ public static bool TryGetAbsoluteIndex(this Position position, SourceText source
             return TryGetAbsoluteIndex(position.Character, position.Line, sourceText, logger, out absoluteIndex);
         }
 
+        public static int GetRequiredAbsoluteIndex(this Position position, SourceText sourceText, ILogger logger)
```

Comment:
Would like @ryanbrandenburg to comment on this. I think `TryGetAbsoluteIndex` was introduced, and deliberately annoying to call, in order to track down a bug where something was throwing an exception, so this might be re-introducing some mystery.

Reply:
I don't remember the specifics of why we made `TryGetAbsoluteIndex`. This seems fine to me.